lib/Interpreter/CppInterOp.cpp

```diff
@@ -3307,22 +3307,42 @@ namespace Cpp {
     return PI->getNameAsString();
   }
 
-  void GetBinaryOperator(TCppScope_t scope, enum BinaryOperator op,
-                         std::vector<TCppFunction_t>& operators) {
-    Decl* D = static_cast<Decl*>(scope);
-    auto* DC = llvm::dyn_cast<DeclContext>(D);
-    Scope* S = getSema().getScopeForContext(DC);
-    if (!S)
-      return;
-
-    clang::UnresolvedSet<8> lookup;
+  OperatorArity getOperatorArity(TCppFunction_t op) {
+    Decl* D = static_cast<Decl*>(op);
+    if (auto* FD = llvm::dyn_cast<FunctionDecl>(D)) {
+      if (FD->isOverloadedOperator()) {
+        switch (FD->getOverloadedOperator()) {
+#define OVERLOADED_OPERATOR(Name, Spelling, Token, Unary, Binary,            \
+                              MemberOnly)                                      \
+    case OO_##Name:                                                            \
+      if ((Unary) && (Binary))                                                 \
+        return kBoth;                                                          \
+      if (Unary)                                                               \
+        return kUnary;                                                         \
+      if (Binary)                                                              \
+        return kBinary;
+#include "clang/Basic/OperatorKinds.def"
+        default:
+          break;
+        }
+      }
+    }
+    return kNone;
+  }
 
-    getSema().LookupBinOp(S, SourceLocation(), (clang::BinaryOperatorKind)op,
-                          lookup);
+  void GetOperator(TCppScope_t scope, Operator op,
+                   std::vector<TCppFunction_t>& operators, OperatorArity kind) {
+    Decl* D = static_cast<Decl*>(scope);
+    if (auto* DC = llvm::dyn_cast_or_null<DeclContext>(D)) {
+      ASTContext& C = getSema().getASTContext();
+      DeclContextLookupResult Result =
+          DC->lookup(C.DeclarationNames.getCXXOperatorName(
+              (clang::OverloadedOperatorKind)op));
 
-    for (NamedDecl* D : lookup) {
-      if (auto* FD = llvm::dyn_cast<Decl>(D))
-        operators.push_back(FD);
+      for (auto* i : Result) {
+        if (kind & getOperatorArity(i))
```

Comment:
```suggestion
        if (kind == getOperatorArity(i))
```

Reply:
When `kind = kBoth` should we return `Unary` $\cap$ `Binary` or `Unary` $\cup$ `Binary`?
This logic using `&` returns `Unary` $\cup$ `Binary`.
Whereas `==` will return   `Unary` $\cap$ `Binary`.

Operators like `-` is both `Unary` and `Binary`.

I set the `enum OperatorArity` values so that I can perform this bitwise operation.

Reply:
When I say kBoth, I expect all unary and binary operators to be present in that list.

Reply:
In that case, we should be using `&`. `kUnary = 0b001` and `kBinary = 0b010` and `kBoth = kUnary ^ kBinary = 0b011`.
No changes are required here.